chore: optimization debian package manager tweaks

rajpratik71 opened the pull request:
By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

Rajpratik71 commented:
@blowmage @lukesneeringer @dazuma @jskeet  requesting review

jskeet commented:
This isn't a repository I have any experience with. I defer to those who use it.

Rajpratik71 commented:
@busunkim96 requesting review

busunkim96 reviewed (approved):
@dazuma @chingor13 @bcoe @codyoss 

Could you confirm that the changes are OK for your language?

(If the images here aren't being used, please let me know and we can remove them to avoid confusion.)

codyoss commented:
Go does not currently use this repository to manage its docker images used for builds(although maybe it should in the future). I defer to others.

dazuma reviewed (changes requested):
Ruby's usage here is complicated, because there are two sources of these images (one here and the other in googleapis/google-cloud-ruby) and they compete with each other and can clobber each other. We need to clean that up, but in the meantime, please do not make any changes to any of the Ruby images in this repo, to prevent this repo from building Ruby images. If you want to make this change to the Ruby images, do so in https://github.com/googleapis/google-cloud-ruby/tree/master/.kokoro/docker instead.

Rajpratik71 commented:
> Ruby's usage here is complicated, because there are two sources of these images (one here and the other in googleapis/google-cloud-ruby) and they compete with each other and can clobber each other. We need to clean that up, but in the meantime, please do not make any changes to any of the Ruby images in this repo, to prevent this repo from building Ruby images. If you want to make this change to the Ruby images, do so in https://github.com/googleapis/google-cloud-ruby/tree/master/.kokoro/docker instead.

done as requested

dazuma commented:
Thanks!

dazuma commented on node/8-puppeteer/Dockerfile:
```diff
@@ -4,16 +4,16 @@ FROM node:8
 # Note: this installs the necessary libs to make the bundled version of Chromium that Puppeteer
 # installs, work.
-RUN apt-get update && apt-get install -y wget --no-install-recommends \
+RUN apt-get update && apt-get install -y --no-install-recommends wget --no-install-recommends \
```
Extra `--no-install-recommends`

Rajpratik71 replied:
fixed

dazuma commented on node/8-puppeteer/Dockerfile:
```diff
@@ -4,16 +4,16 @@ FROM node:8
 
 # See https://crbug.com/795759
-RUN apt-get update && apt-get install -yq libgconf-2-4
+RUN apt-get update && apt-get install -y --no-install-recommendsq libgconf-2-4
```
Bad copy-paste

Rajpratik71 replied:
fixed

dazuma commented on node/12-puppeteer/Dockerfile:
```diff
@@ -4,16 +4,16 @@ FROM node:12
 # Note: this installs the necessary libs to make the bundled version of Chromium that Puppeteer
 # installs, work.
-RUN apt-get update && apt-get install -y wget --no-install-recommends \
+RUN apt-get update && apt-get install -y --no-install-recommends wget --no-install-recommends \
```
Extra `--no-install-recommends`

Rajpratik71 replied:
fixed

dazuma commented on node/12-puppeteer/Dockerfile:
```diff
@@ -4,16 +4,16 @@ FROM node:12
 
 # See https://crbug.com/795759
-RUN apt-get update && apt-get install -yq libgconf-2-4
+RUN apt-get update && apt-get install -y --no-install-recommendsq libgconf-2-4
```
Bad copy-paste

Rajpratik71 replied:
fixed

JustinBeckwith commented:
@chingor13 - can I trouble you to give this one a review and decide if it's a change we want?

chingor13 reviewed (approved):
This looks fine to me.

parthea commented:
Hi @Rajpratik71 , Please could you resolve conflicts ?

parthea commented:
/gcbrun

parthea commented:
@Rajpratik71 The `conventionalcommits.org` check is failing. Please can you take a look? Changing the commit message to `chore: optimization debian package manager tweaks` may resolve the issue.

rajpratik71 committed:
chore: optimization debian package manager tweaks

By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

Rajpratik71 commented:
> @Rajpratik71 The `conventionalcommits.org` check is failing. Please can you take a look? Changing the commit message to `chore: optimization debian package manager tweaks` may resolve the issue.

@parthea  done as required

parthea commented:
/gcbrun

tmatsuo commented:
So are we pushing docker images for testing without knowing our tests would pass or not?
I would say this again and again. This is a recipe for disaster :(

parthea commented:
I'm going to close this PR as there is potential risk that downstream tests will fail and the risk is not worth the benefit of smaller downloads.
